### ISSUE TITLE
Issue Tappx Adapter

### ISSUE THREAD

**Issue by Ruben-Tappx** [prebid/prebid-server-java#1263]
Hello,

My name is Rubén and I am a IT Team Lead in Tappx.

We have issues with our adapter in your solution, from April 23 all the traffic that we received from your solution disappeared and the publishers didn't stop our bidder....

Can you investigate if there is something wrong in our adapter? I saw that you made some changes on April 21... too much of a coincidence...

Thanks

**Comment by RodionOrets** [comment]
@Ruben-Tappx Hi, looking into the issue
Looks like this commit is the root of issue: https://github.com/prebid/prebid-server-java/commit/909af56b4a3defe25119c6ba962c68155c0a4e15
This change was done to prebid-server-java due to same change in prebid-server: https://github.com/prebid/prebid-server/pull/1777

**Comment by RodionOrets** [comment]
@Ruben-Tappx We did rollouts on 8th and 27th of April. Nothing was released in-between these dates. So no changes on 23rd of April as well. Please provide some more info if that's possible

**Comment by bretg** [comment]
@Ruben-Tappx - I do see that responses from your endpoint stop on April 27th, which corresponds to our release of PBS-Java 1.62.

There were two tappx-related pull requests in that release:

- Tappx changes - Backward compatible change of version https://github.com/prebid/prebid-server-java/pull/1222
- Tappx User Syncer + Site Update https://github.com/prebid/prebid-server-java/pull/1171

We're getting errors from the endpoint:
- "Connection refused: ssp.api.tappx.com/35.188.250.24:443"
- "Unexpected status code: 404. Run with request.test = 1 for more info"

Here's a sample request being sent
```
            "uri": "https://ssp.api.tappx.com/ZZ33959IS?tappxkey=pub-54765-ios-6974&v=1.2&type_cnn=prebid",
            "requestbody": "{\"id\":\"7b3e93dc-10cc-42cd-b855-76ddea2e3f7d\",\"imp\":[{\"id\":\"test-div\",\"banner\":{\"format\":[{\"w\":300,\"h\":250}]},\"bidfloor\":0.05,\"secure\":0,\"ext\":{\"bidder\":{\"host\":\"ssp.api.tappx.com/rtb/v2/\",\"tappxkey\":\"pub-54765-ios-6974\",\"endpoint\":\"ZZ33959IS\",\"bidfloor\":0.05}}}],\"site\":{\"domain\":\"rubitest.com\",\"page\":\"http://rubitest.com/index.html\",\"publisher\":{\"id\":\"18534\"},\"ext\":{\"amp\":0}},\"device\":{\"ua\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36\",\"geo\":{\"country\":\"us\"},\"ip\":\"20.14.1.24\"},\"test\":1,\"at\":1,\"tmax\":1000,\"cur\":[\"USD\"],\"source\":{\"tid\":\"7b3e93dc-10cc-42cd-b855-76ddea2e3f7d\"},\"ext\":{\"prebid\":{\"targeting\":{\"pricegranularity\":{\"precision\":2,\"ranges\":[{\"max\":20,\"increment\":0.1}]},\"includewinners\":true,\"includebidderkeys\":true},\"cache\":{\"bids\":{}},\"channel\":{\"name\":\"web\"}}}}",
            "responsebody": "<html>\n<head><title>404 Not Found</title></head>\n<body bgcolor=\"white\">\n<center><h1>404 Not Found</h1></center>\n<hr><center>nginx</center>\n</body>\n</html>\n<!-- a padding to disable MSIE and Chrome friendly error page -->\n<!-- a padding to disable MSIE and Chrome friendly error page -->\n<!-- a padding to disable MSIE and Chrome friendly error page -->\n<!-- a padding to disable MSIE and Chrome friendly error page -->\n<!-- a padding to disable MSIE and Chrome friendly error page -->\n<!-- a padding to disable MSIE and Chrome friendly error page -->\n\n",
            "status": 404
          }
```

I note that the "host" parameter doesn't seem to be seen by the adapter -- it's hitting ssp.api.tappx.com and not ssp.api.tappx.com/rtb/v2/

My guess is that's the problem.

@RodionOrets - please take a look in the adapter to see why the host parameter isn't seen. I opened internal ticket HB-12208 to track this.

**Comment by Ruben-Tappx** [comment]
Hello guys,

The problem is in the "uri"... the correct is "ssp.api.tappx.com/rtb/v2/"

**Comment by bretg** [comment]
Fixed in PBS-Java 1.64. Released and confirmed in the Magnite environment. Will add create the open source release tomorrow.

**Comment by Ruben-Tappx** [comment]
Thanks, we are receiving again the traffic!
